modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2Sender.java

```diff
@@ -106,6 +109,19 @@ public static void sendOn(EndpointDefinition endpoint,
      * @param smc the Synapse message context sent as the response
      */
     public static void sendBack(org.apache.synapse.MessageContext smc) {
+
+        API api = (API) smc.getProperty(RESTConstants.PROCESSED_API);
```

Comment:
Can this break when sending back in a proxy?

Reply:
no, casting a null does not throw an error. We are checking for null in the following line